tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Controllers/EmailController.cs

```diff
@@ -95,86 +102,35 @@ public IActionResult DownloadAttachmentsAsStream([FromBody]MailAttachmentDetails
                 #endregion 
                 Stream downloadAttachments = documentProvision.DownloadAttachments(mailAttachmentDetails);
 
-                return matterCenterServiceFunctions.ServiceResponse(downloadAttachments, (int)HttpStatusCode.OK);
-            }
-            catch (Exception ex)
-            {
-                customLogger.LogError(ex, MethodBase.GetCurrentMethod().DeclaringType.Name, MethodBase.GetCurrentMethod().Name, logTables.SPOLogTable);
-                throw;
-            }
-        }
+                var fileContentResponse = new HttpResponseMessage(HttpStatusCode.OK);
+                fileContentResponse.Headers.Clear();
 
+                fileContentResponse.Content = new StreamContent(downloadAttachments);
 
-        /// <summary>
-        /// For document attachment
-        /// </summary>
-        /// <param name="mailAttachmentDetails"></param>
-        /// <returns></returns>
-        [HttpPost("downloadattachments")]
-        [Produces(typeof(HttpResponseMessage))]
-        [SwaggerOperation("downloadAttachments")]
-        [SwaggerResponse(HttpStatusCode.OK, Description = "Returns IActionResult is of type HttpResponseMessage", Type = typeof(HttpResponseMessage))]
-        [SwaggerResponseRemoveDefaults]
-        public IActionResult DownloadAttachments([FromBody]MailAttachmentDetails mailAttachmentDetails)
-        {
-            try
-            {
-                
-                #region Error Checking                
-                GenericResponseVM genericResponse = null;
-                if (mailAttachmentDetails == null && mailAttachmentDetails.AttachmentContent == null)
-                {
-                    genericResponse = new GenericResponseVM()
-                    {
-                        Value = errorSettings.MessageNoInputs,
-                        Code = HttpStatusCode.BadRequest.ToString(),
-                        IsError = true
-                    };
-                    return matterCenterServiceFunctions.ServiceResponse(genericResponse, (int)HttpStatusCode.OK);
-                }
-                #endregion 
-                string emailText = WebUtility.UrlDecode(mailAttachmentDetails.AttachmentContent);
-                MemoryStream emailStream = GenerateStreamFromString(emailText);
-                string emailName = documentSettings.TempEmailName + new Guid().ToString() + ServiceConstants.EMAIL_FILE_EXTENSION;
-                HttpResponseMessage result = new HttpResponseMessage(HttpStatusCode.OK);                
-                result.Content = new StreamContent(emailStream);
-                result.Content.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue("application/octet-stream");
-                result.Content.Headers.Add("content-disposition", "inline; filename=" + emailName);
+                fileContentResponse.Content.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue(DocumentProvision.ReturnExtension(string.Empty));
```

Comment:
HttpResponseMessage has been constructed and sending that HttpResponseMessage as a response. This HttpResponseMessage  contains complete attachments information that clients wants to download